File: backend/places-service/internal/modules/place/domain/place.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type PlaceStatus string

const (
	StatusDraft             PlaceStatus = "draft"
	StatusPendingModeration PlaceStatus = "pending_moderation"
	StatusPublished         PlaceStatus = "published"
	StatusRejected          PlaceStatus = "rejected"
	StatusArchived          PlaceStatus = "archived"
)

type Category struct {
	ID        uuid.UUID `json:"id"`
	Slug      string    `json:"slug"`
	Title     string    `json:"title"`
	Icon      string    `json:"icon"`
	Color     string    `json:"color"`
	SortOrder int       `json:"sort_order"`
	Active    bool      `json:"active"`
	CreatedAt time.Time `json:"created_at"`
}

type Place struct {
	ID           uuid.UUID   `json:"id"`
	Title        string      `json:"title"`
	Description  string      `json:"description"`
	Latitude     float64     `json:"latitude"`
	Longitude    float64     `json:"longitude"`
	Address      *string     `json:"address"`
	City         *string     `json:"city"`
	Country      *string     `json:"country"`
	AuthorID     *uuid.UUID  `json:"author_id"`
	Status       PlaceStatus `json:"status"`
	RejectReason *string     `json:"reject_reason"`
	Source       string      `json:"source"`
	CoverMediaID *uuid.UUID  `json:"cover_media_id"`
	CategoryIDs  []uuid.UUID `json:"category_ids"`
	PublishedAt  *time.Time  `json:"published_at"`
	CreatedAt    time.Time   `json:"created_at"`
	UpdatedAt    time.Time   `json:"updated_at"`
}

type BBox struct {
	SouthLat float64
	WestLng  float64
	NorthLat float64
	EastLng  float64
}

type ListFilter struct {
	BBox         *BBox
	CategorySlug string
	Q            string
	Limit        int
}

File: landmark_app/lib/features/map/domain/entities/place.dart
enum PlaceVisibility { public, private }

class Place {
  final String id;
  final String title;
  final double latitude;
  final double longitude;
  final String? description;
  final String? city;
  final String? country;
  final String? coverMediaId;
  final List<String> categorySlugs;
  final PlaceVisibility visibility;

  const Place({
    required this.id,
    required this.title,
    required this.latitude,
    required this.longitude,
    this.description,
    this.city,
    this.country,
    this.coverMediaId,
    this.categorySlugs = const [],
    this.visibility = PlaceVisibility.public,
  });

  bool get isPrivate => visibility == PlaceVisibility.private;

  factory Place.fromJson(Map<String, dynamic> j) => Place(
        id: j['id'] as String,
        title: j['title'] as String,
        latitude: (j['latitude'] as num).toDouble(),
        longitude: (j['longitude'] as num).toDouble(),
        description: j['description'] as String?,
        city: j['city'] as String?,
        country: j['country'] as String?,
        coverMediaId: j['cover_media_id'] as String?,
        categorySlugs:
            (j['category_slugs'] as List<dynamic>?)?.cast<String>() ?? [],
        visibility: (j['visibility'] as String?) == 'private'
            ? PlaceVisibility.private
            : PlaceVisibility.public,
      );
}

class PlaceCategory {
  final String id;
  final String slug;
  final String title;
  final String icon;
  final String color;

  const PlaceCategory({
    required this.id,
    required this.slug,
    required this.title,
    required this.icon,
    required this.color,
  });

  factory PlaceCategory.fromJson(Map<String, dynamic> j) => PlaceCategory(
        id: j['id'] as String,
        slug: j['slug'] as String,
        title: j['title'] as String,
        icon: j['icon'] as String? ?? '',
        color: j['color'] as String? ?? '#0E7C7B',
      );
}
